docs/developer-feature-tutorial.md

```diff
@@ -99,74 +165,113 @@ Now, run zulip terminal and open web app from a different account. Start composi
+* The `user_id` of the sender is present in the PM narrow of the user.
 
-If all the above conditions are satisfied we can successfully update the footer to display `X is typing` until we receive
-a `stop` event for typing.
+If all the above conditions are satisfied, we can successfully update the footer to display `<sender> is typing...` until we receive a `stop` event for typing.
 
-To check for the above conditions, we create a function in `ui.py`:
-```python
+To check for the above conditions, we create a method in the Model class in `model.py`:
 
-    def handle_typing_event(self, event: Dict['str', Any]) -> None:
+```python
+# zulipterminal/model.py
+# In Model class
+def _handle_typing_event(self, event: Event) -> None:
+    """
+    Handle typing notifications (in private messages)
+    """
+    assert event['type'] == "typing"
+    if hasattr(self.controller, 'view'):
         # If the user is in pm narrow with the person typing
-        if len(self.model.narrow) == 1 and\
-                self.model.narrow[0][0] == 'pm_with' and\
-                event['sender']['email'] in self.model.narrow[0][1].split(','):
+        narrow = self.narrow
+        if (len(narrow) == 1 and narrow[0][0] == 'pm_with' and event['sender']['email'] in narrow[0][1].split(',')):
             if event['op'] == 'start':
-                user = self.model.user_dict[event['sender']['email']]
-                self._w.footer.set_text([
+                user = self.user_dict[event['sender']['email']]
+                self.controller.view.set_footer_text([
                     ' ',
                     ('code', user['full_name']),
                     ' is typing...'
                 ])
-                self.controller.update_screen()
             elif event['op'] == 'stop':
-                self._w.footer.set_text(self.get_random_help())
-                self.controller.update_screen()
+                self.controller.view.set_footer_text()
+            else:
+                raise RuntimeError("Unknown typing event operation")
 ```
-If the conditions are satisfied, we display `x is typing` if the `op` is `start` and display help message if the `op` is `stop`.
+
+The above code uses the`set_footer_text()` method implemented in the `View` class in `ui.py`, which should default to setting a random Help message when not being used.
+
+If the conditions are satisfied, we display `<sender> is typing...` if the `op` is `start` and display a help message if the `op` is `stop`.
 
 There are two parts to updating a widget in urwid:
-* Changing the widget
-* Updating the screen
 
-This line of code,
-```python
-self._w.footer.set_text([
-                    ' ',
-                    ('code', user['full_name']),
-                    ' is typing...'
-                ])
-```
-changes the footer text, and this
+* Changing the widget, which is done by View
+* Updating the screen, which is done by Controller
+
+These are done by the highlighted lines accordingly in `set_footer_text`
```

Comment:
I think we should avoid coupling the documentation to this method (ie. not include the controller code) - we can advise the reader to look for that function in that file, to see what it does, if they're interested - and also observe the change+update_screen divide perhaps?